deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java

```diff
@@ -237,6 +247,53 @@ List<HotDeploymentWatchedFileBuildItem> hotDeploymentWatchedFiles(QuinoaConfig q
         return watchedFiles;
     }
 
+    private void autoDetectFramework(Path packageJsonFile, QuinoaConfig quinoaConfig, LaunchModeBuildItem launchMode) {
```

Comment:
Not sure why, but mutating parameters is always bugging me, I am pretty sure you to @melloware as I begin to see you also like clean things :)

So I guess you considered other options, returning a new instance means, having to assign all other fields which might also cause errors, so I follow you on this.

Still, maybe it should be clear in the method name, so I would split this in two methods:
- `detectFramework`
- 'initDefaultConfig(config, framework)`

And make sure all the mutations are as close as possible. I mean no other logic in between beside the empty checks.

WDYT?

Reply:
OK let me see if I can rework this a little bit!

Reply:
Refactored into two distinct methods.

Reply:
So far in all my testing that has been the case...but I am by **NO** means a Quarkus threading or build expert. 😄

Reply:
@geoand might know about 1 and 2.

Reply:
I am pretty sure that there is no guarantee that build steps run in the same thread - IIRC we allow to the processing graph to be completed via multiple threads, so classes that contain `@BuildStep` should not contain any state.